src/sagas.js

```diff
@@ -250,16 +245,25 @@ export function* refreshMainPage ({ shallowFetch, page }) {
   yield put(updateVmsPoolsCount())
 }
 
-function* refreshDetailPage ({ id }) {
+function* refreshDetailPage ({ id, onNavigation, onSchedule }) {
   yield selectVmDetail(actionSelectVmDetail({ vmId: id }))
   yield getConsoleOptions(actionGetConsoleOptions({ vmId: id }))
+
+  // Load ISO images on manual refresh click
+  if (!onNavigation && !onSchedule) {
+    yield fetchIsoFiles(getIsoFiles())
+  }
 }
 
-function* refreshDialogPage ({ id }) {
+function* refreshDialogPage ({ id, onNavigation, onSchedule }) {
```

Comment:
Is `onNavigation` and `onSchedule` really needed here? Because dialog page must up-to-date all the time even with ISO files.

Reply:
ISO file loading is potentially a very heavy operation.  Doing it every minute doesn't seem like a good idea.  Doing it once at login and then allowing it to be refreshed manually (without a page reload) feels like a good trade-off.

Reply:
Plus I'd assume the set of ISO are going to be relatively static and not in need of frequent refreshes.

Reply:
Indeed, ISOs are very rarely changed. If it came down to it, I'd be ok with telling someone to logout and back in to get a new one that the admin had just loaded :) We won't do that because we do have a refresh button in admin portal. But, tl;dr, caching++ on this one

Reply:
**Note**: I updated the ISO caching/refresh scheme for this PR such that ISOs are loaded at login time, and then only refreshed when a user manually click refresh on a detail (view or dashboard) or dialog (edit) page.